src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.AbstractRemoveDocumentUndoUnit.cs

```diff
@@ -12,13 +13,16 @@ internal partial class VisualStudioWorkspaceImpl
         private abstract class AbstractRemoveDocumentUndoUnit : AbstractAddRemoveUndoUnit
         {
             protected readonly DocumentId DocumentId;
+            protected readonly IEnumerable<string> CreatedFolder;
```

Comment:
please don't hold onto IENumerables.  It's not clear if they're lazy, and what thread they may be computed on.  Use an IReadOnlyList or ImmutableArray.